src/libratbag-private.h

```diff
@@ -131,6 +131,20 @@ struct ratbag_driver {
 	void (*remove)(struct ratbag_device *device);
 
 	/**
+	 * Callback called when the driver should write any profiles that
+	 * were modified back to the device.
+	 *
+	 * Both profile and button structs have a dirty variable that can
+	 * be used to tell whether or not they've actually changed since
+	 * the last commit. In order to reduce the amount of time
+	 * committing takes, drivers should use this information to avoid
+	 * writing back profiles and buttons that haven't actually changed.
+	 *
+	 * Drivers need not modify dirty, as ratbag will do this for you.
```

Comment:
skip the dirty bit explanation, it's an implementation detail and since drivers don't need to handle it it shouldn't be mentioned